src/spinner/spinner.js

```diff
@@ -15,133 +15,93 @@
-   * Store strings for class names defined by this component that are used in
-   * JavaScript. This allows us to simply change it in one place should we
-   * decide to modify at a later date.
+   * Attach all listeners to the DOM.
    *
-   * @enum {string}
-   * @private
+   * @export
    */
-  MaterialSpinner.prototype.CssClasses_ = {
-    MDL_SPINNER_LAYER: 'mdl-spinner__layer',
-    MDL_SPINNER_CIRCLE_CLIPPER: 'mdl-spinner__circle-clipper',
-    MDL_SPINNER_CIRCLE: 'mdl-spinner__circle',
-    MDL_SPINNER_GAP_PATCH: 'mdl-spinner__gap-patch',
-    MDL_SPINNER_LEFT: 'mdl-spinner__left',
-    MDL_SPINNER_RIGHT: 'mdl-spinner__right'
-  };
+  addEventListeners() {
+    // No event listeners needed for MaterialSpinner.
+  }
 
   /**
-   * Auxiliary method to create a spinner layer.
+   * Remove all listeners from the DOM.
    *
-   * @param {number} index Index of the layer to be created.
-   * @public
+   * @export
    */
-  MaterialSpinner.prototype.createLayer = function(index) {
-    var layer = document.createElement('div');
-    layer.classList.add(this.CssClasses_.MDL_SPINNER_LAYER);
-    layer.classList.add(this.CssClasses_.MDL_SPINNER_LAYER + '-' + index);
-
-    var leftClipper = document.createElement('div');
-    leftClipper.classList.add(this.CssClasses_.MDL_SPINNER_CIRCLE_CLIPPER);
-    leftClipper.classList.add(this.CssClasses_.MDL_SPINNER_LEFT);
-
-    var gapPatch = document.createElement('div');
-    gapPatch.classList.add(this.CssClasses_.MDL_SPINNER_GAP_PATCH);
-
-    var rightClipper = document.createElement('div');
-    rightClipper.classList.add(this.CssClasses_.MDL_SPINNER_CIRCLE_CLIPPER);
-    rightClipper.classList.add(this.CssClasses_.MDL_SPINNER_RIGHT);
-
-    var circleOwners = [leftClipper, gapPatch, rightClipper];
-
-    for (var i = 0; i < circleOwners.length; i++) {
-      var circle = document.createElement('div');
-      circle.classList.add(this.CssClasses_.MDL_SPINNER_CIRCLE);
-      circleOwners[i].appendChild(circle);
-    }
-
-    layer.appendChild(leftClipper);
-    layer.appendChild(gapPatch);
-    layer.appendChild(rightClipper);
-
-    this.element_.appendChild(layer);
-  };
-  MaterialSpinner.prototype['createLayer'] =
-      MaterialSpinner.prototype.createLayer;
+  removeEventListeners() {
+    // No event listeners needed for MaterialSpinner.
+  }
 
   /**
-   * Stops the spinner animation.
-   * Public method for users who need to stop the spinner for any reason.
+   * Set whether or not the spinner should be active.
    *
-   * @public
+   * @param {boolean} value The value to set the property to.
+   * @export
    */
-  MaterialSpinner.prototype.stop = function() {
-    this.element_.classList.remove('is-active');
-  };
-  MaterialSpinner.prototype['stop'] = MaterialSpinner.prototype.stop;
+  set active(value) {
+    if (value) {
+      this.root_.classList.add(MaterialSpinner.cssClasses_.IS_ACTIVE);
+    } else {
+      this.root_.classList.remove(MaterialSpinner.cssClasses_.IS_ACTIVE);
+    }
```

Comment:
Unfortunately, we can't do that if we're targeting IE11, as [it doesn't support the second parameter in the `toggle` method](http://caniuse.com/#search=classList) :(